app/__init__.py

```diff
@@ -38,5 +44,49 @@ def new_token_route():
                 500,
             )
 
+    def Captcha_detection(image):
+
+    # write to disk
+    with open('temppiratecaptcha.png','wb') as file:
+        file.write(image.read())
+
+    curr = time.time()
+    result = get_captcha_string('temppiratecaptcha.png')
+    print('detect_image done, elapsed: ', str(time.time() - curr ), ' result: ', result)
+    return result
+
+    @app.route('/ikagod/ikabot', methods=['POST'])
+    def home():
+    try:
+        if 'upload_file' in request.files:
+            threadqueue.append(threading.current_thread().ident)
+            print(threading.active_count())
+            while True:
+                with lock:
+                    if threading.current_thread().ident == threadqueue[-1]:
+                        captcha = Captcha_detection(request.files['upload_file'])
```

Comment:
Before processing uploaded files, be sure to validate their type, size, etc. Avoid accepting just any type of file, and limit the size allowed. Using Flask-Uploads can be useful for managing uploaded files more securely

Reply:
I intend to remove the need to save to disk completely. The image will be directly loaded from the bytes object returned by image.read() in the Captcha_detection function. Checks for image validity, size and dimensions will also be made in the same function.